lib/ui/recommendation.js

```diff
@@ -48,16 +104,25 @@ Recommendation.prototype = {
       this.win.setTimeout(this._pollForElement, 20);
     }
   },
+
   show: function(data) {
-    // assign the data to this.data
-    // render using a simple inline template string
-    // unset this.el.collapsed to show the el
-    // fire 'recommendation-shown' event
+    const recommendation = new RecommendationRow(this.win, data);
+    this.win.universalSearch.console.log('recommendation.show', recommendation, recommendation.render().outerHTML);
+    this.data = data;
+    this.el.appendChild(recommendation.render());
+    this.el.collapsed = false;
+    this.events.publish('recommendation-shown', this);
   },
+
   hide: function() {
-    // collapse the node
-    // null out this.data
+    this.win.universalSearch.console.log('recommendation.hide');
+    this.el.collapsed = true;
+    while (this.el.firstChild) {
```

Comment:
It should be enough to just pop the row element off of `this.el`, as the other nodes are all child nodes of the row. This works too, but it should only ever run once.